packages/message-relayer/test/unit-tests/relay-tx.spec.ts

```diff
@@ -0,0 +1,375 @@
+import { expect } from '../setup'
+
+/* Imports: External */
+import hre from 'hardhat'
+import { Contract, Signer } from 'ethers'
+import { getContractFactory } from '@eth-optimism/contracts'
+import { smockit } from '@eth-optimism/smock'
+import { toPlainObject } from 'lodash'
+
+/* Imports: Internal */
+import {
+  makeRelayTransactionData,
+  getStateRootBatchByTransactionIndex,
+  getStateBatchAppendedEventByTransactionIndex,
+  getMessageByTransactionHash,
+} from '../../src/relay-tx'
+
+describe('relay transaction generation functions', () => {
+  const ethers = (hre as any).ethers
+  const l1RpcProvider = ethers.provider
+  const l2RpcProvider = ethers.provider
+
+  let signer1: Signer
+  before(async () => {
+    ;[signer1] = await ethers.getSigners()
+  })
+
+  let MockL2CrossDomainMessenger: Contract
+  beforeEach(async () => {
```

Comment:
Any reason for multiple `beforeEach` blocks?

Reply:
Mostly just for readability.